### ISSUE TITLE
[BUG][RUBY and others] Default parameters should not be added in code generation

### ISSUE THREAD

**Issue by davidgm0** [OpenAPITools/openapi-generator#17257]
#### Bug Report Checklist

- [ ] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

Reading about [parameters](https://swagger.io/docs/specification/describing-parameters/), the `default` is:

>  The default value is the one that the server uses if the client does not supply the parameter value in the request

As I understand, this means that default does not have to be added by the client. However, code generation adds the default value to the request. I would expect that this is not added by the client. I believe this is not specific to Ruby (the one I tried this with)

##### openapi-generator version

openapi-generator-cli 7.2.0-SNAPSHOT
  commit : 46dd875

##### OpenAPI declaration file content or url

```json
{
  "openapi": "3.0.3",
  "info": {
    "title": "Sample API",
    "version": "1.0.0"
  },
  "paths": {},
  "components": {
    "schemas": {
      "ExampleObject": {
        "type": "object",
        "properties": {
          "state": {
            "type": "string",
            "enum": [
              "foo",
              "bar"
            ],
            "default": "foo",
            "description": "The state of the object."
} } } } } }
```

##### Generation Details

```
❯ mkdir config
❯ touch config/dummy.json
# paste the definition into dummy.json
❯ docker run --rm \
                                             -v "$(PWD)/config/:/local" \
                                             -v "$(PWD):/out" \
                                            --pull always \
                                             openapitools/openapi-generator-cli generate \
                                               --skip-validate-spec \
                                               --generator-name ruby --output /out \
                                               --input-spec /local/dummy.json \
                                               --additional-properties gemName=dummy
```

##### Steps to reproduce

```ruby
# lib/probely/models/example_object.rb

      if attributes.key?(:'state') 
        self.state = attributes[:'state']
      else
        self.state = 'foo' # Line 86
      end
```

I think this `else` should not be there

##### Related issues/PRs

There was this old discussion [here](https://github.com/OpenAPITools/openapi-generator/issues/2260)

##### Suggest a fix

In the description

**Comment by jschmitz-sofi** [comment]
I agree; this causes issues in mobile development, where a new client would need to be generated and released through the app store to make a default value adjustment. The default value should be able to be adjusted in the spec and backend without requiring an update to the client.
